src/components/Carousel.vue

```diff
@@ -0,0 +1,276 @@
+            && entries.length > 0
+            && !entries.some((entry) => typeof entry !== 'string'),
+    })
+    public entries!: string[];
+
+    @Prop(String)
+    public selected?: string;
+
+    @Prop({
+        type: Number,
+        default: 16,
+    })
+    public entryMargin!: number;
+
+    @Prop({
+        type: Number,
+        default: 1000,
+    })
+    public animationDuration!: number; // in ms
+
+    @Prop({
+        type: Boolean,
+        default: false,
+    })
+    public hideBackgroundEntries!: boolean;
+
+    @Prop({
+        type: Boolean,
+        default: false,
+    })
+    public disabled!: boolean;
+
+    public $refs: { [ref: string]: HTMLElement[] }; // these are arrays because of v-for
+
+    private effectiveSelected: string = '';
+    private radius: Tweenable = new Tweenable();
+    private rotations: Map<string, Tweenable> = new Map(); // map entry -> rotation
+    private requestAnimationFrameId: number | null = null;
+
+    private get _hasDummyPosition(): boolean {
+        // add dummy to avoid that second entry is hidden exactly behind selected item on opposite side of circle.
+        return this.entries.length <= 2;
+    }
+
+    private get _totalPositionCount(): number {
+        return this.entries.length + (this._hasDummyPosition ? 1 : 0);
+    }
+
+    private async mounted() {
+        this._onKeydown = this._onKeydown.bind(this);
+        document.addEventListener('keydown', this._onKeydown);
+        // trigger these manually instead of via immediate watcher to avoid animating on first render
+        await this._updateDimensions(false);
+        this._updateSelection(this.selected);
+        this._updateRotations(false);
+    }
+
+    private destroyed() {
+        document.removeEventListener('keydown', this._onKeydown);
+        if (this.requestAnimationFrameId === null) return;
+        cancelAnimationFrame(this.requestAnimationFrameId);
+    }
+
+    @Watch('entries')
+    private async _onEntriesChange() {
+        await this._updateDimensions();
+        this._updateSelection(this.effectiveSelected); // revalidate
+        this._updateRotations();
+    }
+
+    @Watch('selected')
+    private _updateSelection(newSelection) {
+        const oldSelection = this.effectiveSelected;
+        const isNewSelectionValid = this.entries.indexOf(newSelection) !== -1;
+        const isOldSelectionValid = this.entries.indexOf(oldSelection) !== -1;
+        if (isNewSelectionValid) {
+            this.effectiveSelected = newSelection;
+        } else if (!isOldSelectionValid) {
+            this.effectiveSelected = this.entries[0];
+        } // else keep the old selection
+
+        if (this.effectiveSelected !== oldSelection) {
+            this.$emit('select', this.effectiveSelected);
+        }
+    }
+
+    @Watch('entryMargin')
+    private async _updateDimensions(newWatcherValueOrTween = true) {
+        const tween = typeof newWatcherValueOrTween === 'boolean' ? newWatcherValueOrTween : true;
```

Comment:
I think this accomplishes the same:
```suggestion
        const tween = newWatcherValueOrTween !== false;
```

Reply:
This method can be called with a boolean and also gets invoked on changes to `entryMargin` with a number.
Therefore the check for `'boolean'` is intentional.
I'll add a type annotation to make it more obvious that it can be a number too.